pkg/server/plugin/keymanager/awskms/awskms_test.go

```diff
@@ -0,0 +1,706 @@
+package awskms
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/aws/aws-sdk-go-v2/service/kms/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/spiffe/spire/pkg/server/plugin/keymanager"
+	"github.com/spiffe/spire/proto/spire/common/plugin"
+	"github.com/spiffe/spire/test/spiretest"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	// Defaults used for testing
+	validAccessKeyID     = "AKIAIOSFODNN7EXAMPLE"
+	validSecretAccessKey = "secret"
+	validRegion          = "us-west-2"
+	kmsKeyID             = "abcd-fghi"
+	kmsAlias             = "alias/SPIRE_SERVER_KEY/spireKeyID"
+	spireKeyID           = "spireKeyID"
+)
+
+var (
+	ctx = context.Background()
+)
+
+func TestKeyManager(t *testing.T) {
+	suite.Run(t, new(KmsPluginSuite))
+}
+
+type KmsPluginSuite struct {
```

Comment:
We've been shifting away from Suite-based tests for a while now. They have proved problematic. Here are a few reasons:
1) Easy to forget to reset state
2) Do not work well with subtests
3) Are hard to provide configurable test setup, i.e., SetupTest() has to be one-size-fits-all, which means many times you end up doing a "second stage" setup in each test case to handle the unique situation for that case.
4) Do not play well with the standard `go test` filters, i.e., you have to use `-testify.m` or whatever (and even that has problems with subtests)

I think this code would clean up quite a bit if you just used standard `func TestFoo(t *testing.T)` test methods. They could call function that would to do the setup currently done by SetupTest(). Then you wouldn't need the `reset` method, and all of your subtests would work properly.

Reply:
Let me know if the last changes fit into this.